src/argilla/client/feedback/dataset/remote/filtered.py

```diff
@@ -92,4 +94,24 @@ def __init__(
         )
 
     def delete(self) -> None:
-        raise NotImplementedError("`delete` does not work for filtered datasets.")
+        raise NotImplementedError(
+            "`delete` does not work for filtered datasets. First call `filter_reset()` and then `delete()`."
+        )
```

Comment:
I think we should be showing `warnings.warn` instead of raising `NotImplementedError` exceptions, as that may break the code and the method will never be implemented, so maybe we're just safer with a `warnings.warn`

Reply:
Plus I'm not sure about `filter_reset()` as mentioned below

Reply:
I removed `filter_reset`